client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/RetryOnFailureOperation.java

```diff
@@ -69,7 +69,6 @@ protected RetryOnFailureOperation(short requestCode, short responseCode, Codec c
 
    @Override
    public void invoke(Channel channel) {
-      assert channel.isActive();
```

Comment:
There is no point in checking if the channel is active upfront since it may become inactive at any moment during transmission. Having that assertion outside the try/catch does not properly trigger the retry logic.

Reply:
@danberindei I've added some javadoc on executeOperation